python/extensions.bzl

```diff
@@ -127,3 +152,59 @@ pip = module_extension(
         "parse": tag_class(attrs = _pip_parse_ext_attrs()),
     },
 )
+
+def _host_hub_impl(repo_ctx):
+    repo_ctx.file("BUILD.bazel", """\
+package(default_visibility = ["//visibility:public"])
+exports_files(["python"])""")
+
+    # use the same functions that we have to
+    # create the toolchains with, that create
+    # the Label "interpreter".
+    (os, arch) = get_host_os_arch(repo_ctx)
+    host_platform = get_host_platform(os, arch)
+
+    found = False
+    for config_label in repo_ctx.attr.interpreters:
+        if host_platform in str(config_label):
+            # We found the interpreter, so we create the symlink
+            # Regardless of the OS, we need a symlink named "Python"
+            # as we have code in the pip rules that is hardcoded on the
+            # name.
+            repo_ctx.symlink(config_label, "python")
+            found = True
+            break
+
+    if not found:
+        fail("Unable to find interpreter, that matches the host os and architecture, in the argument interpreters")
+
+""" 
+    This repository rule finds the host's Python interpreter and creates a symlink to it.
+
+    We use this rule to set the pips interpreter target when using different Operating 
+    Systems with the same project.
+"""
+host_hub = repository_rule(
+    implementation = _host_hub_impl,
+    local = True,
+    attrs = dict({
+        "interpreters": attr.label_list(
+            mandatory = True,
+            doc = """\
+This rule find the compatible interpreter for the host and creates a repo that includes a 
+symlink to the interpreters Python binary.
+""",
+        ),
+        "repo": attr.label(
```

Comment:
Looks unused?

Reply:
Yah. I used it previously.  Will remove.

Reply:
Done